components/accelerator/nvidia/query/detect.go

```diff
@@ -0,0 +1,145 @@
+package query
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/leptonai/gpud/log"
+	"github.com/leptonai/gpud/pkg/file"
+	"github.com/leptonai/gpud/pkg/process"
+
+	"github.com/NVIDIA/go-nvlib/pkg/nvlib/device"
+	nvinfo "github.com/NVIDIA/go-nvlib/pkg/nvlib/info"
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+)
+
+// Returns true if the local machine has NVIDIA GPUs installed.
+func GPUsInstalled(ctx context.Context) (bool, error) {
+	smiInstalled := SMIExists()
+	if !smiInstalled {
+		return false, nil
+	}
+	log.Logger.Debugw("nvidia-smi installed")
+
+	// now that nvidia-smi installed,
+	// check the NVIDIA GPU presence via PCI bus
+	pciDevices, err := ListPCIs(ctx)
```

Comment:
it is only useful for printing?

Reply:
Renamed to `ListNVIDIAPCIs`. It's only listing the devices with NVIDIA name in it, thus checking whether the host has NVIDIA devices.